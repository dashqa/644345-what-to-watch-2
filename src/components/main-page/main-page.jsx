import React, {Fragment} from "react";
import PropTypes from "prop-types";
import TopPage from "../top-page/top-page.jsx";
import BottomPage from "../bottom-page/bottom-page.jsx";

const MainPage = ({movies}) => {
  return (
    <Fragment>
      <TopPage/>
      <BottomPage movies={movies}/>
    </Fragment>
  );
};

MainPage.propTypes = {
  movies: PropTypes.array.isRequired,
};

export default MainPage;
