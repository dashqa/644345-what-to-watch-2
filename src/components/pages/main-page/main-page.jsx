import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";

import {getFilteredMovies, getPromoMovie} from "@store/movies-data/selectors";

import Header from "@partials/header/header";
import MovieCardBackground from "@partials/movie-card-background/movie-card-background";
import MovieCardPoster from "@partials/movie-card-poster/movie-card-poster";
import MovieCardInfo from "@partials/movie-card-info/movie-card-info";
import MovieButtons from "@partials/movie-buttons/movie-buttons";
import Catalog from "@partials/catalog/catalog";
import Footer from "@partials/footer/footer";

const MainPage = ({movies, promoMovie}) => {
  const {id, name, posterImage, backgroundImage, genre, released, isFavorite} = promoMovie;
  return (
    <>
      <section className="movie-card">
        <MovieCardBackground
          name={name}
          backgroundImage={backgroundImage}
        />
        <h1 className="visually-hidden">WTW</h1>

        <Header
          classMods="movie-card__head"
          needUserBlock
        />

        <div className="movie-card__wrap">
          <div className="movie-card__info">

            <MovieCardPoster
              name={name}
              posterImage={posterImage}
            />

            <div className="movie-card__desc">
              <MovieCardInfo
                name={name}
                genre={genre}
                released={released}
              />
              <MovieButtons
                movieId={id}
                isFavorite={isFavorite}
              />
            </div>
          </div>
        </div>
      </section>

        <div className="page-content">
          <Catalog
            movies={movies}
            headerClassMods="visually-hidden"
            sectionTitle="Catalog"
            needFilter
            needShowMore
          />

          <Footer/>
        </div>
    </>
  );
};

MainPage.defaultProps = {
  movies: [],
  promoMovie: {},
};

MainPage.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    posterImage: PropTypes.string,
    previewImage: PropTypes.string,
    backgroundImage: PropTypes.string,
    backgroundColor: PropTypes.string,
    description: PropTypes.string,
    rating: PropTypes.number,
    scoresCount: PropTypes.number,
    director: PropTypes.string,
    starring: PropTypes.arrayOf(PropTypes.string),
    runTime: PropTypes.number,
    genre: PropTypes.string,
    released: PropTypes.number,
    isFavorite: PropTypes.bool,
    videoLink: PropTypes.string,
    previewVideoLink: PropTypes.string
  })).isRequired,
  promoMovie: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    posterImage: PropTypes.string,
    previewImage: PropTypes.string,
    backgroundImage: PropTypes.string,
    backgroundColor: PropTypes.string,
    description: PropTypes.string,
    rating: PropTypes.number,
    scoresCount: PropTypes.number,
    director: PropTypes.string,
    starring: PropTypes.arrayOf(PropTypes.string),
    runTime: PropTypes.number,
    genre: PropTypes.string,
    released: PropTypes.number,
    isFavorite: PropTypes.bool,
    videoLink: PropTypes.string,
    previewVideoLink: PropTypes.string
  }).isRequired,
};

const mapStateToProps = (state) => ({
  movies: getFilteredMovies(state),
  promoMovie: getPromoMovie(state),
});

export {MainPage};

export default connect(mapStateToProps)(MainPage);
