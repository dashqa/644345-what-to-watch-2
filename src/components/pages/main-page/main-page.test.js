import React from 'react';
import renderer from "react-test-renderer";
import {MainPage} from './main-page';
import {BrowserRouter} from "react-router-dom";

const props = {
  movies: [],
  promoMovie: {}
};

jest.mock(`@partials/catalog/catalog.jsx`, () => `Catalog`);
jest.mock(`@partials/header/header.jsx`, () => `Header`);
jest.mock(`@partials/movie-buttons/movie-buttons.jsx`, () => `MovieButtons`);

it(`Main page correctly renders after relaunch`, () => {
  const tree = renderer.create(<BrowserRouter><MainPage {...props}/></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
