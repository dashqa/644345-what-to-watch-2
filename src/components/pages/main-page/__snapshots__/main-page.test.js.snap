// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page correctly renders after relaunch 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt=""
        src=""
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <Header />
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt=""
            height="327"
            src=""
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              
            </span>
            <span
              className="movie-card__year"
            >
              0
            </span>
          </p>
          <MovieButtons />
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <Catalog
      isMainPage={true}
      movies={Array []}
    />
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
